server/src/test/java/org/apache/kafka/server/share/ShareSessionCacheTest.java

```diff
@@ -0,0 +1,113 @@
+        ShareSessionKey key5 = cache.maybeCreateSession("grp", Uuid.randomUuid(), 410, mockedSharePartitionMap(50));
+        assertNull(key5);
+    }
+
+    @Test
+    public void testResizeCachedSessions() {
+        ShareSessionCache cache = new ShareSessionCache(2, 100);
+        assertEquals(0, cache.size());
+        assertEquals(0, cache.totalPartitions());
+        ShareSessionKey key1 = cache.maybeCreateSession("grp", Uuid.randomUuid(), 0, mockedSharePartitionMap(2));
+        assertNotNull(key1);
+        assertShareCacheContains(cache, new ArrayList<>(Collections.singletonList(key1)));
+        ShareSession session1 = cache.get(key1);
+        assertEquals(2, session1.size());
+        assertEquals(2, cache.totalPartitions());
+        assertEquals(1, cache.size());
+
+        ShareSessionKey key2 = cache.maybeCreateSession("grp", Uuid.randomUuid(), 0, mockedSharePartitionMap(4));
+        assertNotNull(key2);
+        assertShareCacheContains(cache, new ArrayList<>(Arrays.asList(key1, key2)));
+        ShareSession session2 = cache.get(key2);
+        assertEquals(6, cache.totalPartitions());
+        assertEquals(2, cache.size());
+        cache.touch(session1, 200);
+        cache.touch(session2, 200);
+
+        ShareSessionKey key3 = cache.maybeCreateSession("grp", Uuid.randomUuid(), 200, mockedSharePartitionMap(5));
+        assertNull(key3);
+        assertShareCacheContains(cache, new ArrayList<>(Arrays.asList(key1, key2)));
+        assertEquals(6, cache.totalPartitions());
+        assertEquals(2, cache.size());
+        cache.remove(key1);
+        assertShareCacheContains(cache, new ArrayList<>(Collections.singletonList(key2)));
+        assertEquals(1, cache.size());
+        assertEquals(4, cache.totalPartitions());
+
+        Iterator<CachedSharePartition> iterator = session2.partitionMap().iterator();
+        iterator.next();
+        iterator.remove();
+        assertEquals(3, session2.size());
+        assertEquals(4, session2.cachedSize());
+        cache.touch(session2, session2.lastUsedMs());
+        assertEquals(3, cache.totalPartitions());
+    }
+
+    private ImplicitLinkedHashCollection<CachedSharePartition> mockedSharePartitionMap(int size) {
+        ImplicitLinkedHashCollection<CachedSharePartition> cacheMap = new
+                ImplicitLinkedHashCollection<>(size);
+        for (int i = 0; i < size; i++)
+            cacheMap.add(new CachedSharePartition("test", Uuid.randomUuid(), i, false));
+        return cacheMap;
+    }
+
+    public void assertShareCacheContains(ShareSessionCache cache,
+                                         ArrayList<ShareSessionKey> sessionKeys) {
+        int i = 0;
+        assertEquals(sessionKeys.size(), cache.size());
+        for (ShareSessionKey sessionKey : sessionKeys) {
+            i++;
+            assertFalse(cache.get(sessionKey).isEmpty(),
+                    "Missing session " + i + " out of " + sessionKeys.size() + " ( " + sessionKey + " )");
```

Comment:
nit: you can ignore as well
```suggestion
            assertFalse(cache.get(sessionKey).isEmpty(),
                    "Missing session " + ++i + " out of " + sessionKeys.size() + " ( " + sessionKey + " )");
```